src/crimson/os/seastore/cached_extent.h

```diff
@@ -181,8 +278,16 @@ class CachedExtent : public boost::intrusive_ref_counter<
    * references in the the buffer with the passed
    * record_block_offset record location.
    */
-  virtual void on_delta_write(paddr_t record_block_offset) {}
+  void on_delta_write(paddr_t record_block_offset) {
+    on_delta_commit(record_block_offset);
+    touched_by = 0;
```

Comment:
This really should happen at the same time as the change to the CachedExtent state -- see my other comment, it's probably worth removing this and doing the updates directly in Cache where the state updates happen.